presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -970,6 +970,24 @@ public void revokeTablePrivileges(Session session, QualifiedObjectName tableName
         return ImmutableList.copyOf(grantInfos.build());
     }
 
+    @Override
+    public boolean isWritingStagingFilesEnabled(Session session)
+    {
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, session.getCatalog().orElseThrow(() -> new IllegalStateException("No catalog specified")));
+        ConnectorId connectorId = catalogMetadata.getConnectorId();
+        ConnectorMetadata metadata = getMetadata(session, connectorId);
+        return metadata.isWritingStagingFilesEnabled(session.toConnectorSession(connectorId));
+    }
+
+    @Override
+    public void commitStagingFiles(Session session, SchemaTableName tableName, Collection<Slice> fragments)
```

Comment:
I would have some handle instead of `SchemaTableName` that holds some information about the current insert (there could be multiple inserts into the same table at the same time). I understand that this information can be included into `Collection<Slice> fragments`. But i would limit the amount of information in the `fragment`. As fragments are collected overt the network from every worker node. So including more information into fragment means more network traffic on a coordinator.